api/src/main/java/org/hawkular/inventory/api/model/MetricType.java

```diff
@@ -157,7 +155,7 @@ private Update() {
 
         public Update(Map<String, Object> properties, MetricUnit unit) {
             super(properties);
-            this.unit = unit;
+            this.unit = unit == null ? MetricUnit.NONE : unit;
```

Comment:
There are three points to this:
- granularity of updates (does the client need to supply all fields or are only the supplied ones updated)
- should we ever allow null as a valid value for MetricUnits
- are there fields like unit that are probably read-only (after first setting)? If a client can change them at will, what are the actual semantics of that? Will e.g. the metrics all update too? What would be the result of a change between incompatible units (e.g. time to kilobytes)